src/update_agent/mod.rs

```diff
@@ -79,14 +79,16 @@ pub struct InteractiveSession {
     tty_dev: String,
 }
 
-/// Function to deserialize field to `Option<String>`, where empty strings are
-/// deserialized into `None`.
-fn empty_string_as_none<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+/// Function to deserialize field to `Option<String>`, where empty strings or
+/// `n/a` (not available) strings are deserialized into `None`. In systemd v254+
+/// loginctl list-sessions --json started outputting `n/a` instead of an empty
+/// string for tty.
+fn empty_or_na_string_as_none<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
```

Comment:
I'd call this "deserialize_systemd_tty_canonicalized" perhaps.  It's only usable for the systemd tty I suspect; I can't imagine many other things use "n/a".  (Or perhaps I'm wrong and systemd started emitting n/a for other things...)

Reply:
updated!